src/Register/index.test.js

```diff
@@ -1,18 +1,91 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { BrowserRouter as Router } from 'react-router-dom';
+import { MockedProvider } from '@apollo/client/testing';
+import AuthProvider from '../AuthProvider';
 import Register from './index';
+import QUERY_USERS from '../graphql/queries';
+import REGISTER from '../graphql/mutations';
 
-it('renders the register page', () => {
-  render(<Router><Register /></Router>);
-  expect(screen.getByRole('form')).toBeInTheDocument();
-  userEvent.type(screen.getByLabelText('Username'), 'User');
+const mockedUsersResponse = {
+  request: { query: QUERY_USERS },
+  result: {
+    data: {
+      users: [
+        {
+          username: 'User',
+          password: 'password',
+          id: 1,
+        },
+      ],
+    },
+  },
+};
+
+const mockedRegisterResponse = {
+  request: {
+    query: REGISTER,
+    variables: {
+      username: 'User_',
+      firstName: 'John',
+      lastName: 'Smith',
+      password: 'password',
+    },
+  },
+  result: {
+    data: {
+      id: 1,
+    },
+  },
+};
+
+const mocks = [
+  mockedUsersResponse,
+  mockedUsersResponse,
```

Comment:
Same as the mock for login, this duplication doesn't feel like a realistic mock of real data, please see if adding data to `users` would work like my other similar suggestion